config/routes.rb

```diff
@@ -4,13 +4,34 @@
   root to: 'projects#index'
 
   namespace :api do
-    resources :deploys, only: [:index, :active_count] do
+    resources :deploys, only: [:index] do
       collection do
         get :active_count
       end
     end
 
-    resources :stages, only: [:index] do
+    resources :deploy_groups, only: [:index]
+
+    resources :projects, param: :project_id, only: [] do
+      member do
+        get :template, to: 'stages#template'
```

Comment:
even if they did, it wouldn't be project/id/stages/template, it would be project/id/stages/stage_id